app/src/features/rules/screens/rulesList/components/RulesList/components/RuleSelectionListDrawer/RuleSelectionListDrawer.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Drawer, DrawerProps } from "antd";
+import { RuleSelectionList, RuleSelectionListProps } from "../RuleSelectionList/RuleSelectionList";
+import "./RuleSelectionListDrawer.scss";
+
+interface RuleSelectionListDrawerProps extends Omit<RuleSelectionListProps, "premiumIconSource"> {
+  open: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+  drawerPlacement?: DrawerProps["placement"];
+  premiumIconSource?: RuleSelectionListProps["premiumIconSource"];
+}
+
+export const RuleSelectionListDrawer: React.FC<RuleSelectionListDrawerProps> = ({
+  open,
+  onClose,
+  groupId,
+  children,
+  drawerPlacement = "right",
+  source,
+  premiumIconSource = "rule_drawer",
+  premiumPopoverPlacement = "topLeft",
+  callback = () => {},
+  onRuleItemClick = () => {},
+}) => {
+  return (
+    <>
+      <Drawer
```

Comment:
Please add `destroyOnClose` prop here. There are some views where there are muliple touchpoints for new rule, which will mount multiple drawer components in the DOM, making it heavy. (eg. rules table).